### DISCUSSION TITLE
which TSS  does Signac use for a gene if a gene has multiple TSSs

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#1293] [Q&A]
Hi Signac team,
Thanks for developing the Signac.
I used a gtf file to annotate the peak in the Seurat object. I am wondering which TSS  does Signac use to annotate a peak if a gene has multiple TSSs? Similarly, which transcript  does Signac use to annotate a peak if a gene has multiple transcripts?

Thanks
Best

**Comment by timoast** [comment]
What do you mean by "annotate a peak" exactly? Which function?

**Reply by pengxin2019** [reply]
I used the code below to annotate my Seurat object.
`Annotation(Seurat.obj) <- gtf
`
Speaking of the annotation of the peaks, I was thinking of the function `ClosestFeature`. 
In the documentation here: https://stuartlab.org/signac/reference/closestfeature, how did you calculate the column `distance`? Are you talking about the distance between who and who?

Also, in the documentation here: https://stuartlab.org/signac/articles/cicero.html, speaking of the `gene track` in the plot created by `CoveragePlot`, how do you create this gene track? if a gene has multiple transcripts in the gtf file, which transcript does Signac use to create this gene track?

Thanks
Best

**Reply by timoast** [reply]
> I used the code below to annotate my Seurat object.
Annotation(Seurat.obj) <- gtf

This adds the information in `gtf` to the Seurat object, it does not modify that information at all

> Speaking of the annotation of the peaks, I was thinking of the function ClosestFeature.
In the documentation here: https://stuartlab.org/signac/reference/closestfeature, how did you calculate the column distance? Are you talking about the distance between who and who?

The `ClosestFeature` function will compute the distance from each peak to the closest feature in the set of gene annotations provided, and report that distance. The `distanceToNearest` function from GenomicRanges is used: https://rdrr.io/bioc/GenomicRanges/man/nearest-methods.html

> Also, in the documentation here: https://stuartlab.org/signac/articles/cicero.html, speaking of the gene track in the plot created by CoveragePlot, how do you create this gene track? if a gene has multiple transcripts in the gtf file, which transcript does Signac use to create this gene track?

The longest transcript is used. Alternatively, all transcript isoforms can be shown by setting `annotation="transcript"`

**Reply by pengxin2019** [reply]
Thanks for promot response.

I was trying to set annotation="transcript"

```
 p <- CoveragePlot(
  object = integrated,
  region = "2-151108982-151113383",#
  extend.upstream = 20000,
  extend.downstream = 20000,
  show.bulk = TRUE,
  ranges = ranges.show,
  region.highlight = ranges.show,
  annotation = "transcript"
)
```
But I got this error: 
Error in if (annotation) { : argument is not interpretable as logical

Here is the description of my Seurat object `integrated`
```
> integrated
An object of class Seurat 
130479 features across 15626 samples within 2 assays 
Active assay: peaks (110444 features, 110272 variable features)
 1 other assay present: RNA
 3 dimensional reductions calculated: integrated_lsi, umap, lsi
```
Can you give me any further comments?

**Reply by timoast** [reply]
Please open an issue and include the output of `sessionInfo()` and your full code
